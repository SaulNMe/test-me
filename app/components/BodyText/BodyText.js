import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { Fonts, Colors } from 'qualificame-native/app/styles';
import styles from './BodyTextStyle';

export default class BodyText extends PureComponent {
	render() {
		const textColor = {color: Colors[this.props.color]};
		const weight = {fontWeight: Fonts.weight[this.props.weight] };
		const align = {textAlign: this.props.align };

		return (
			<View>
				<Text style={[styles.text, textColor, weight, align]}>{this.props.text}</Text>
			</View>
		);
	}
}

	BodyText.propTypes = {
		text: PropTypes.string,
		color: PropTypes.string,
		weight: PropTypes.string,
		align: PropTypes.string,
	}

	BodyText.defaultProps = {
		text: "",
		color: "dark",
		weight: "regular",
		align: "left"
	}
