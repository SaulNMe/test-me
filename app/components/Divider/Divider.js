import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { Colors  } from 'qualificame-native/app/styles';

import styles from './DividerStyle';

export default class Divider extends PureComponent {
  render() {
	return (
		<View style={[styles.divider, {backgroundColor: Colors[this.props.color]}]} />
	);
  }	
}

Divider.propTypes = {
	color: PropTypes.string
}

Divider.defaultProps = {
	color: 'gray'
}
