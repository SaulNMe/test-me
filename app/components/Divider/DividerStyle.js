import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	divider: {
		height: 1,
		color: Colors.light,
		width:'100%'
	}
});
