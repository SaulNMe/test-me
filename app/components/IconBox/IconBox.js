import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { LinearGradient } from 'expo-linear-gradient';

import styles from './IconBoxStyle';
import { Feather } from '@expo/vector-icons';
import { Colors } from 'qualificame-native/app/styles';

export default class IconBox extends PureComponent {
	render() {
		let gradientColors;
		switch(this.props.bgColor) {
			case 'green':
				gradientColors = ['#0F9F92', '#12B184'];
			break;
			case 'blue':
				gradientColors = ['#154BBA', '#107EC2'];
			break;
			case 'orange':
				gradientColors = ['#E62808', '#FC680A'];
			break;
			case 'pink':
				gradientColors = ['#E31862', '#EB194D'];
			break;
			case 'gray':
				gradientColors = ['#ccc', '#ccc'];
			break;
			default:
				gradientColors = ['#fff', '#fff'];
			break;
		}
		return (
			<LinearGradient colors={gradientColors} style={[styles.defaultSize, styles.centerObjects]} start={[0, 0]} end={[1, 0]}>
				<Feather name={this.props.iconName} size={24} color={Colors.white}/>
			</LinearGradient>
		);
	}
}

	IconBox.propTypes = {
		bgColor: PropTypes.string,
		iconName: PropTypes.string

	}

	IconBox.defaultProps = {
		bgColor: 'green',
		iconName: 'x'
	}
