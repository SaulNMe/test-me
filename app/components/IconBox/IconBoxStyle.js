import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.flexBox,
	defaultSize:{
		width: 32,
		height: 32,
		borderRadius: 5
	}
});
