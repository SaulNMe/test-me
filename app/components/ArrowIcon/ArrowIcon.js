import React, { PureComponent } from 'react';
import { View, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import IconBox from 'qualificame-native/app/components/IconBox';
import styles from './ArrowIconStyle';
import BodyText from 'qualificame-native/app/components/BodyText';
import { Feather } from '@expo/vector-icons';
import { Colors } from 'qualificame-native/app/styles';
import Divider from 'qualificame-native/app/components/Divider';

export default class ArrowIcon extends PureComponent {
	render() {
		return (
			<TouchableOpacity
				onPress={this.props.onPress}
				activeOpacity={0.6}
				style={styles.bgColor}
			>
				{ this.props.addTopDivider && <Divider /> }	
				<View style={[styles.row, styles.baseHorizontalMargin, styles.alignItemsCenter, styles.smallVerticalPadding]}>
					<IconBox 
						bgColor={this.props.bgColor}	
						iconName={this.props.iconName}
					/>
					<View style={[styles.smallHorizontalMargin, styles.flex1]}>
						<BodyText 
							text={this.props.text}
							color={this.props.color}
							weight='light'
						/>
					</View>
					{ !this.props.removeIcon && <Feather name="chevron-right" size={24} color={Colors.dark}/>}
				</View>
				{ this.props.addBottomDivider && <Divider /> }	
			</TouchableOpacity>
		);
	}
}

	ArrowIcon.propTypes = {
		onPress: PropTypes.func,
		bgColor: PropTypes.string,
		iconName: PropTypes.string,
		text: PropTypes.string,
		addTopDivider: PropTypes.bool,
		addBottomDivider: PropTypes.bool,
		removeIcon: PropTypes.bool,
		color: PropTypes.string,
	}

	ArrowIcon.defaultProps = {
		onPress: ()=>{},
		bgColor: 'green',
		iconName: 'x',
		text: '',
		addBottomDivider: false,
		addTopDivider: false,
		removeIcon: false,
		color: 'dark'
	}
