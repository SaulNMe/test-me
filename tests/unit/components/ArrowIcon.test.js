import React from 'react';
import { shallow } from 'enzyme';

import ArrowIcon from 'qualificame-native/app/components/ArrowIcon';
import BodyText from 'qualificame-native/app/components/BodyText';
import Divider from 'qualificame-native/app/components/Divider';
import IconBox from 'qualificame-native/app/components/IconBox';
import { Text, TouchableOpacity } from 'react-native';
import { Feather } from '@expo/vector-icons';


function createTestProps (props) {
	return {
		// common props
		createdAtText: 'created',
		text: 'text test',
		addTopDivider: false,
		addBottomDivider: false,
		onPress: ()=>{},
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<ArrowIcon {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('displays the correct text ', () => {
		expect(wrapper.find(BodyText).prop('text')).toEqual('text test');
	});

	it('displays dark BodyText', () => {
		expect(wrapper.find(BodyText).prop('color')).toEqual('dark');
	});

	describe('when addtopDivider is true', () => {
		beforeEach(() => {
			wrapper = createWrapper({ addTopDivider: true });
		});
		it('displays only top divider', () => {
			expect(wrapper.find(Divider).getElements().length).toEqual(1);
			expect(wrapper.find(TouchableOpacity).children().first().name()).toEqual("Divider");
			expect(wrapper.find(TouchableOpacity).children().last().name()).not.toBe("Divider");
		});
	});

	describe('when addBottomDivider is true', () => {
		beforeEach(() => {
			wrapper = createWrapper({ addBottomDivider: true });
		});
		it('displays only bottom divider', () => {
			expect(wrapper.find(Divider).getElements().length).toEqual(1);
			expect(wrapper.find(TouchableOpacity).children().last().name()).toEqual("Divider");
			expect(wrapper.find(TouchableOpacity).children().first().name()).not.toBe("Divider");
		});
	});	

	describe('when addTopDivider and addBottomDivider are true', () => {
		beforeEach(() => {
			wrapper = createWrapper({ addBottomDivider: true, addTopDivider: true });
		});
		it('displays both dividers', () => {
			expect(wrapper.find(Divider).getElements().length).toEqual(2);
			expect(wrapper.find(TouchableOpacity).children().last().name()).toEqual("Divider");
			expect(wrapper.find(TouchableOpacity).children().first().name()).toEqual("Divider");
		});
	});

	describe('when props are not set', () => {
		beforeEach(() => {
			wrapper = createWrapper();
		});
		
		it('displays no text', () => {
			expect(wrapper.find(BodyText).prop('text')).toBeFalsy();
		});
		
		it('dividers are disabled', () => {
			expect(wrapper.find(Divider).getElements().length).toEqual(0);
		});

		it('displays iconBox', () => {
			expect(wrapper.find(IconBox).exists()).toBeTruthy();
		});		

		it('displays feather icon', () => {
			expect(wrapper.find(Feather).exists()).toBeTruthy();
		});
	});
});

describe('interactions', () => {
	let wrapper;
	const createWrapper = props => shallow(<ArrowIcon {...props}/>);
	describe('when onPress function is passed to props', () => {
		let fn;
		beforeEach(() => {
			fn = jest.fn();
			const props = createTestProps({ onPress: fn })
			wrapper = createWrapper(props);
		});
		it('it is called on button press', () => {
			wrapper.find(TouchableOpacity).first().props().onPress();
			expect(fn).toBeCalled();
		});
	});
})
