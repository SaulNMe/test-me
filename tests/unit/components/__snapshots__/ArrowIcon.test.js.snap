// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.6}
  onPress={[Function]}
>
  <Component
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "marginHorizontal": 16,
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "paddingVertical": 8,
        },
      ]
    }
  >
    <IconBox
      bgColor="green"
      iconName="x"
    />
    <Component
      style={
        Array [
          Object {
            "marginHorizontal": 8,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <BodyText
        align=""
        color="dark"
        text="text test"
        weight=""
      />
    </Component>
    <Icon
      allowFontScaling={false}
      color="#364350"
      name="chevron-right"
      size={24}
    />
  </Component>
</TouchableOpacity>
`;
